src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -69,6 +71,21 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
         return userChallengesSet;
     }
 
+    @Override
+    public List<ChallengeDTO> getChallengesByCreatorId(Long creatorId) throws NotFoundException {
+
+        List<Challenge> challenges = challengeRepository.findAllByCreatorId(creatorId);
+
```

Comment:
In cazul acesta de GET, ar trebui sa arunci exceptie doar daca nu gasesti Challenge-uri dupa creatorId. Prin urmare, ar trebui sa verifici daca List<Challenge> are valori, in caz contrar arunci exceptie. Altfel faci conversia la list de DTO si o returnezi. Exceptiile ar trebui verificate si aruncate cat mai repede, astfel incat sa salvam o executie inutila, cum ar fi in cazul tau liniile 79, 81. De asemenea, acum nu trebuie sa va complicati cu Optional si sa-l vedeti ca pe singura solutie. Puteti face si if-uri clasice. Ce vreau sa spun este ca nu ar trebui sa faceti mecanisme de genul Optional.ofNullable doar ca sa faceti aruncare de exceptie folosind acel orElseThrow. Incercam sa ne folosim de Optional, acolo unde este necesar, altfel poti merge pe versiunea clasica.